delphi-checks/src/main/java/au/com/integradev/delphi/checks/EmptyRoutineCheck.java

```diff
@@ -42,6 +44,18 @@ public DelphiCheckContext visit(RoutineImplementationNode routine, DelphiCheckCo
     return super.visit(routine, context);
   }
 
+  @Override
+  public DelphiCheckContext visit(AnonymousMethodNode anonymousMethod, DelphiCheckContext context) {
+    if (anonymousMethod.isEmpty() && shouldAddViolation(anonymousMethod)) {
```

Comment:
It seems odd to have `anonymousMethod.isEmpty()`, one of the two criteria for whether a violation should be added, outside of `shouldAddViolation`. Could we refactor so that this is inside `shouldAddViolation`? Same with the existing `RoutineImplementationNode` code.

Reply:
Done.